tests/spicy/types/unit/self-recursive.spicy

```diff
@@ -0,0 +1,14 @@
+# @TEST-EXEC: spicyc -j -d -o test.hlto %INPUT
+# @TEST-EXEC: ${SCRIPTS}/printf '\x00' | spicy-dump test.hlto >>output
+# @TEST-EXEC: ${SCRIPTS}/printf '\x01\x00' | spicy-dump test.hlto >>output
```

Comment:
This seems to work well, but I think it would be great to add a test with one more level of nesting given that in the impl we just use a boolean `top_level`, e.g.,

```
# @TEST-EXEC: ${SCRIPTS}/printf '\x01\x01\x00' | spicy-dump test.hlto >>output
```